src/lib/create/from-array.js

```diff
@@ -1,19 +1,16 @@
 import { hooks } from '../utils/hooks';
-import { createDate, createUTCDate } from './date-from-array';
+import { createUTCDate } from './date-from-array';
 import { daysInYear } from '../units/year';
 import { weekOfYear, weeksInYear, dayOfYearFromWeeks } from '../units/week-calendar-utils';
 import { YEAR, MONTH, DATE, HOUR, MINUTE, SECOND, MILLISECOND } from '../units/constants';
 import { createLocal } from './local';
 import defaults from '../utils/defaults';
 import getParsingFlags from './parsing-flags';
 
-function currentDateArray(config) {
+function currentDateArray(zone) {
     // hooks is actually the exported moment object
-    var nowValue = new Date(hooks.now());
-    if (config._useUTC) {
-        return [nowValue.getUTCFullYear(), nowValue.getUTCMonth(), nowValue.getUTCDate()];
-    }
-    return [nowValue.getFullYear(), nowValue.getMonth(), nowValue.getDate()];
+    var nowValue = hooks.withTimeZone(zone)()._d;
```

Comment:
This is the only real change I made to Tim's code. You can see his [here](https://github.com/moment/moment/pull/3134/files#diff-9ef3d17fb73729933240837060eeb3ecR12). I was lucky to have run the tests while my local date was different than UTC's or I'd have never seen it.